src/strongbus.ts

```diff
@@ -690,10 +702,25 @@ export class Bus<TEventMap extends Events.EventMap = Events.EventMap> implements
   private emitLifecycleEvent<L extends Lifecycle>(event: L, payload: Lifecycle.EventMap<TEventMap>[L]): void {
     const handlers = this.lifecycle.get(event);
     if(handlers && handlers.size) {
-      handlers.forEach(async fn => {
+      for(const fn of handlers) {
         try {
-          await fn(payload);
+          const execution = fn(payload);
+          // Emit errors if fn returns promise that rejects
+          (execution as Promise<any>)?.catch?.((e) => {
+            if(event === Lifecycle.error) {
```

Comment:
The only thing I don't like about this PR is the redundancy here. Lines 709-717 are the same as 724-734.

I thought about pulling this error logic into a function to eliminate the redundancy, but I wanted to avoid the overhead of creating a closure when we hit the happy path.